From 1b49e14c82af691fe1ec5aa5de8392350bce11a1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Tue, 2 Feb 2016 11:50:51 +0000
Subject: [PATCH] Resolves: tdf#97375 use Invalidate in all modes

makes the crash/hangs go away

(cherry picked from commit ab5c427784fb72d52042b8122ffc5a0fd7108c6b)

(cherry picked from commit c3f09ae629b349c52a4a7954e3017ceb8d7afeaf)

Change-Id: I91a4391190ec7aa0ffa5e41a8c1eb86b4bb9c484
Reviewed-on: https://gerrit.libreoffice.org/22026
Reviewed-by: Michael Stahl <mstahl@redhat.com>
Tested-by: Michael Stahl <mstahl@redhat.com>
---
 editeng/source/editeng/impedit3.cxx | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/editeng/source/editeng/impedit3.cxx b/editeng/source/editeng/impedit3.cxx
index fb071e0..98e3d51 100644
--- a/editeng/source/editeng/impedit3.cxx
+++ b/editeng/source/editeng/impedit3.cxx
@@ -289,12 +289,7 @@ void ImpEditEngine::UpdateViews( EditView* pCurView )
         {
             // convert to window coordinates ....
             aClipRect = pView->pImpEditView->GetWindowPos( aClipRect );
-
-            // For tiled rendering, we have to always go via Invalidate().
-            if ( pView == pCurView && !pView->isTiledRendering())
-                Paint( pView->pImpEditView, aClipRect, 0, true );
-            else
-                pView->GetWindow()->Invalidate( aClipRect );
+            pView->GetWindow()->Invalidate( aClipRect );
         }
     }
 
-- 
2.5.0

